pallets/authors-manager/src/lib.rs

```diff
@@ -256,6 +256,30 @@ pub mod pallet {
 
             return Ok(())
         }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(Weight::zero())]
+        #[transactional]
+        pub fn rotate_author_ethereum_key(
+            origin: OriginFor<T>,
+            author_account_id: T::AccountId,
+            author_old_eth_public_key: ecdsa::Public,
+            author_new_eth_public_key: ecdsa::Public,
+        ) -> DispatchResult {
+            let _ = ensure_root(origin)?;
+
+            ensure!(
+                !<EthereumPublicKeys<T>>::contains_key(&author_new_eth_public_key),
+                Error::<T>::AuthorEthKeyAlreadyExists
+            );
+
+            let author_id = EthereumPublicKeys::<T>::take(&author_old_eth_public_key)
```

Comment:
Can we check that authorId and old key match?

Reply:
we do, line 278 right after this